### DISCUSSION TITLE
Set default time for date picker

### DISCUSSION THREAD

**Discussion by panchenkodv** [quasarframework/quasar#11550] [Ideas / Proposals] [answered]
I need set default time 23:59 for date picker in case when I use q-date and q-time with q-input. You have posibility to set default date in time picker. I realy need such posibility for setting default time for date picker which differ from 00:00. Thanks!

**Comment by MilosPaunovic** [comment]
Does it need to differ from setting a default date/time in this example from the docs:
[https://quasar.dev/vue-components/time#example--qdate-and-qtime-with-qinput](https://quasar.dev/vue-components/time#example--qdate-and-qtime-with-qinput)

**Reply by panchenkodv** [reply]
Yes, it need to be differ. My v-model comes from server and has value null. I should set default time after user pickup date. This behavior is similar when you set 'default-date' prop and pickup time first.

**Reply by MilosPaunovic** [reply]
Okay, so, is anything stopping you to change the value of the variable to your default value (the one you're using as a `v-model` for input) in the exact time when you need it to be changed?

**Reply by panchenkodv** [reply]
I need to change the time immediately after picking the date. How can I do it?

**Reply by MilosPaunovic** [reply]
So, you get date from server, let's say `01-01-2022` and you want to append that to input field with time of `23:59`?

**Reply by panchenkodv** [reply]
No. I get empty model from server with field date_end = null. And I ask the user to choose the date from picker and append to that value time 23:59.
I have the form filed like this

https://codepen.io/Kroshyk/pen/mdBJYZP

item -- is my empty model from server
P.S. Sorry, can't properly format code in the post

**Comment by pdanpdan** [comment]
Please put this in a codepen

**Reply by panchenkodv** [reply]
Done

**Comment by pdanpdan** [comment, marked as answer]
https://codepen.io/pdanpdan/pen/YzrXoBV?editors=1010

**Reply by panchenkodv** [reply]
Look like hack, but it works, thanks!
Are there any plans in the future to modify the q-date component and allow the default-date property? Because I have 7 similar forms and I need to copy this code into each of them. Not very comfortable(

**Reply by panchenkodv** [reply]
Some developer faced with this issue last year
https://forum.quasar-framework.org/topic/6818/set-default-time-for-date-picker
